# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/style/FontFaceSet.h
# Commit: 3f2605087550
# Full Hash: 3f26050875502a932b4f39326bf0f0de958441ac
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2022-06-29 03:44:30
# Regressor Bug: 1771493
# File Overlap Count: 1
# Description:
#   Bug 1771493 - Part 4. Split FontFaceSetDocumentImpl from FontFaceSetImpl. r=emilio
#   
#   This patch splits document specific functionality from FontFaceSetImpl
#   into a new class FontFaceSetDocumentImpl. This will make it easier to
#   fork FontFaceSetImpl for workers.
# ==============================================================================

diff -r 5e6bf07c9c27 -r 3f2605087550 layout/style/FontFaceSet.h
--- a/layout/style/FontFaceSet.h	Tue Jun 28 14:27:21 2022 +0000
+++ b/layout/style/FontFaceSet.h	Tue Jun 28 14:27:21 2022 +0000
@@ -16,7 +16,7 @@
 
 class nsFontFaceLoader;
 class nsIPrincipal;
-class nsPIDOMWindowInner;
+class nsIGlobalObject;
 struct RawServoFontFaceRule;
 
 namespace mozilla {
@@ -36,7 +36,8 @@
   NS_DECL_ISUPPORTS_INHERITED
   NS_DECL_CYCLE_COLLECTION_CLASS_INHERITED(FontFaceSet, DOMEventTargetHelper)
 
-  FontFaceSet(nsPIDOMWindowInner* aWindow, dom::Document* aDocument);
+  static already_AddRefed<FontFaceSet> CreateForDocument(
+      dom::Document* aDocument);
 
   virtual JSObject* WrapObject(JSContext* aCx,
                                JS::Handle<JSObject*> aGivenProto) override;
@@ -112,6 +113,7 @@
  private:
   friend mozilla::dom::FontFaceSetIterator;  // needs GetFontFaceAt()
 
+  explicit FontFaceSet(nsIGlobalObject* aParent);
   ~FontFaceSet();
 
   /**